node.go

```diff
@@ -147,6 +153,23 @@ func (n *Node) EnableTLSTransport(keyFile string, certFile string, caFile string
 	n.enableTLS = true
 }
 
+// EnableTLSTransportFromText make transport protected by TLS.
+// etcd.pkg.transport.TLSInfo don't support use context as the cert. so
+// this function will create the cert file
+func (n *Node) EnableTLSTransportFromText(key []byte, cert []byte, ca []byte) error {
+	if err := creatAndWriteFile(key, DefaultKeyFileName); err != nil {
```

Comment:
You don't should create file for key, cert and ca, please consider the use `tls.X509KeyPair`.

Reply:
The problem is transport only provides  set up TLS from file path, 
https://github.com/etcd-io/etcd/blob/master/pkg/transport/listener.go#L66

Reply:
If etcd only supports set up TLS from file, it unnecessary that add `EnableTLSTransportFromText`.